lib/semantics/mod-file.cc

```diff
@@ -13,32 +13,43 @@
 // limitations under the License.
 
 #include "mod-file.h"
+#include "resolve-names.h"
 #include "scope.h"
 #include "semantics.h"
 #include "symbol.h"
 #include "../evaluate/traversal.h"
+#include "../parser/message.h"
 #include "../parser/parsing.h"
 #include <algorithm>
 #include <cerrno>
 #include <fstream>
 #include <ostream>
+#include <set>
+#include <string_view>
+#include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <unistd.h>
 #include <vector>
 
 namespace Fortran::semantics {
 
 using namespace parser::literals;
 
-// The initial characters of a file that identify it as a .mod file.
+// The first line of a file that identifies it as a .mod file.
 // The first three bytes are a Unicode byte order mark that ensures
 // that the module file is decoded as UTF-8 even if source files
 // are using another encoding.
-static constexpr auto magic{"\xef\xbb\xbf!mod$ v1 sum:"};
+struct ModHeader {
+  static constexpr const char *bom{"\xef\xbb\xbf"};
+  static constexpr const char *magic{"!mod$ v1 sum:"};
+  static constexpr int magicLen{std::strlen(magic)};
+  static constexpr int sumLen{16};
+  static constexpr char terminator{'\n'};
+  static constexpr int len{magicLen + 1 + sumLen};
```

Comment:
Is the byte order mark no longer included in the length?

Reply:
When writing we need to take into account the byte order mark but when reading using `Prescan` it is not included in the file contents that are returned. So there are two different header lengths to consider.